superset-frontend/src/explore/components/controls/DndColumnSelectControl/DndColumnSelect.tsx

```diff
@@ -59,8 +59,9 @@ export function DndColumnSelect(props: DndColumnSelectProps) {
     }
     if (
       typeof value === 'string' &&
-      typeof optionSelectorValues === 'string' &&
-      value !== optionSelectorValues
+      Array.isArray(optionSelectorValues) &&
+      optionSelectorValues.length === 1 &&
+      value !== optionSelectorValues[0]
```

Comment:
I looked at `OptionSelector.has` - I agree that should probably be changed to something like

```typescript
  has(value: string): boolean {
    return ensureIsArray(this.getValues()).includes(value);
  }
```

Would changing that solve this problem?

Reply:
Done. thanks